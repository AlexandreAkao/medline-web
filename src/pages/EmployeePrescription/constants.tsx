import { UseTableCellProps } from 'react-table';

export const COLUMNS: IColumn[] = [
  {
    Header: 'Lugar na fila',
    accessor: 'statusaa',
    Cell: ({ row }) => {
      const position = String(row.index + 1);
      return position.padStart(4, '0');
    },
  },
  {
    Header: 'Nome',
    accessor: 'createdBy.name',
  },
  {
    Header: 'CPF',
    accessor: 'createdBy.cpf',
  },
  {
    Header: 'Status',
    accessor: 'status',
    Cell: ({ value }: UseTableCellProps<IRequest, RequestStatusType>) => {
      const mapStatus: Record<RequestStatusType, string> = {
        EM_ESPERA: 'Em espera',
        CONCLUIDO: 'Conclu√≠do',
        REJEITADO: 'Rejeitado',
      };

      return mapStatus[value];
    },
  },
];
