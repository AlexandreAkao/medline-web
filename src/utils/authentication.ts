import { isEmpty } from 'ramda';

export const getUserInfo = () => {
  const user = JSON.parse(localStorage.getItem('user') ?? '{}');

  return isEmpty(user) ? null : (user as IUser);
};

export const geToken = () => localStorage.getItem('token');
