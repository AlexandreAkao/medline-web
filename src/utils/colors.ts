export const getColorAlpha = (color: string, alpha = 1) => {
  const alphaLevel = Math.round(255 * alpha).toString(16);
  return color + alphaLevel;
};
