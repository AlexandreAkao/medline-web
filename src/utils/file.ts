export const downloadFile = (blob: Blob, filename: string) => {
  const link = document.createElement('a');
  link.href = URL.createObjectURL(blob);
  link.download = filename;
  link.click();
};
