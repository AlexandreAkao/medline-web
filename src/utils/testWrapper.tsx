import { render } from '@testing-library/react';
import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';

import { AuthProvider } from 'providers/AuthProvider';
import { LoaderProvider } from 'providers/LoaderProvider';

export function RouterProviders({
  children,
  route = '/',
  history = createMemoryHistory({ initialEntries: [route] }),
}: IChildren & { route?: string; history?: MemoryHistory }) {
  return (
    <Router location={history.location} navigator={history}>
      {children}
    </Router>
  );
}

export const renderWithRouter = (
  component: React.ReactNode,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {},
) => {
  return {
    ...render(
      <Router location={history.location} navigator={history}>
        {component}
      </Router>,
    ),
    history,
  };
};

export const renderWithProvider = (
  component: React.ReactNode,
  { route = ['/'], history = createMemoryHistory({ initialEntries: route }) } = {},
) => {
  return {
    ...render(
      <Router location={history.location} navigator={history}>
        <LoaderProvider>
          <AuthProvider>{component}</AuthProvider>,
        </LoaderProvider>
      </Router>,
    ),
    history,
  };
};
