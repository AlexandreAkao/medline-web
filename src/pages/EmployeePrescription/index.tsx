import {
  EmployeePrescriptionContainer,
  EmployeePrescriptionMain,
  EmployeePrescriptionTitle,
} from 'pages/EmployeePrescription/styles';
import MedlineHeader from 'components/Header/MedlineHeader';
import Table from 'components/Table';
import { COLUMNS } from 'pages/EmployeePrescription/constants';

function EmployeePrescription() {
  return (
    <EmployeePrescriptionContainer>
      <MedlineHeader />

      <EmployeePrescriptionMain>
        <EmployeePrescriptionTitle>Atestados e Receitas</EmployeePrescriptionTitle>
        <Table columns={COLUMNS} data={[]} />
      </EmployeePrescriptionMain>
    </EmployeePrescriptionContainer>
  );
}

export default EmployeePrescription;
