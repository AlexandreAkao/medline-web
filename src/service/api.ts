import axios, { AxiosError } from 'axios';
import { toast } from 'react-toastify';

import { getToken } from 'utils/authentication';

const api = axios.create({
  baseURL: (import.meta.env.VITE_BFF_BASE_URL as string) ?? '',
});

api.interceptors.request.use(
  config => {
    const authorization = getToken() ? { Authorization: `Bearer ${getToken()}` } : null;
    return {
      ...config,
      headers: {
        ...authorization,
      },
    };
  },
  error => {
    return Promise.reject(error);
  },
);

api.interceptors.response.use(
  response => {
    return response;
  },
  (error: AxiosError) => {
    if (error.response?.status === 401) {
      axios
        .post<IRefreshToken>(
          'https://securetoken.googleapis.com/v1/token',
          {
            grant_type: 'refresh_token',
            refresh_token: localStorage.getItem('refresh_token') ?? '',
          },
          {
            params: {
              key: 'AIzaSyDYolHTkT0zuI3X1MNup36eSi2gZ3INa2g',
            },
          },
        )
        .then(data => {
          const { access_token: accessToken } = data.data;

          localStorage.setItem('token', accessToken);
        });
    } else if (error.response?.status !== 404)
      if (error.response && error.response.status >= 500)
        toast.error(error.message, { autoClose: 3000, theme: 'colored' });
    return Promise.reject(error);
  },
);

export default api;
