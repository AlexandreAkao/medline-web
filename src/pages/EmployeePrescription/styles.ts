import styled from 'styled-components';

import { TableContainer } from 'components/Table/styles';
import colors from 'styles/colors';

export const EmployeePrescriptionContainer = styled.div`
  background-color: ${colors.white.lighter};
  height: 100%;
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-position-x: 50%;
  overflow: auto;
`;

export const EmployeePrescriptionMain = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 2rem auto;
  width: 80%;

  ${TableContainer} {
    width: 100%;
  }
`;

export const EmployeePrescriptionTitle = styled.h1`
  font-size: 2.125rem;
  font-weight: 500;
  margin-bottom: 2rem;
`;
