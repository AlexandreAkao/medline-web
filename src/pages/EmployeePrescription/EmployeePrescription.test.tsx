import { screen } from '@testing-library/react';
import { vi } from 'vitest';

import { renderWithProvider } from 'utils/testWrapper';
import EmployeePrescription from 'pages/EmployeePrescription';
import api from 'service/api';
import { paginationMock } from '__tests__/mocks/pagination.mock';
import { requestMock } from '__tests__/mocks/request.mock';

describe('EmployeePrescription', () => {
  beforeEach(() => {
    vi.spyOn(api, 'get').mockImplementation(() =>
      Promise.resolve({
        data: paginationMock([requestMock]),
      }),
    );
  });

  const renderComponent = () => {
    return renderWithProvider(<EmployeePrescription />);
  };

  it('should match with snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render correctly', () => {
    renderComponent();
    const titleElement = screen.getByRole('heading');

    expect(titleElement).toHaveTextContent('Atestados e Receitas');
  });
});
