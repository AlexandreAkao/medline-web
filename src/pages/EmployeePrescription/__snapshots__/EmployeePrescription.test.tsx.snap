// Vitest Snapshot v1

exports[`EmployeePrescription > should match with snapshot 1`] = `
<div>
  <div
    class="sc-gsnTZi fatJle"
  >
    <header
      class="sc-jSMfEi ypYGZ"
      data-testid="test-navigation"
    >
      <a
        href="/"
        id="header-logo"
      >
        <img
          alt="Medline logo"
          class="sc-gKXOVf dKYKzm"
          src="/src/assets/images/medline-logo.svg"
        />
      </a>
      <nav
        class="sc-iBkjds iOmqAx"
        data-testid="test-navigation-items"
      >
        <div
          class="sc-eCYdqJ bpawvP"
        >
          <button
            class="sc-papXJ fnEGjm"
            type="button"
          >
            Cadastro
          </button>
        </div>
        <div
          class="sc-eCYdqJ bpawvP"
        >
          <a
            href="/login"
          >
            Entrar
          </a>
        </div>
      </nav>
    </header>
    <div
      class="sc-dkzDqf hTMlLd"
    >
      <h1
        class="sc-hKMtZM cZFLfV"
      >
        Atestados e Receitas
      </h1>
      <div
        class="sc-bczRLJ kzXtbC"
      >
        <table
          role="table"
        >
          <thead>
            <tr
              role="row"
            >
              <th
                colspan="1"
                role="columnheader"
              >
                Lugar na fila
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                Nome
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                CPF
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody
            role="rowgroup"
          />
        </table>
      </div>
      <div
        class="sc-hAZoDl dAlGie"
      >
        <ul
          class="pagination-container"
        >
          <li
            class="previous disabled"
          >
            <a
              aria-disabled="true"
              aria-label="Previous page"
              class=" "
              rel="prev"
              role="button"
              tabindex="-1"
            >
              &lt;
            </a>
          </li>
          <li
            class="next"
          >
            <a
              aria-disabled="false"
              aria-label="Next page"
              class=""
              rel="next"
              role="button"
              tabindex="0"
            >
              &gt;
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  ,
</div>
`;
